python/GafferArnoldTest/IECoreArnoldPreviewTest/RendererTest.py

```diff
@@ -536,6 +536,42 @@ def testOutputFilters( self ) :
 			self.assertEqual( arnold.AiNodeEntryGetName( arnold.AiNodeGetNodeEntry( f ) ), "gaussian_filter" )
 			self.assertEqual( arnold.AiNodeGetFlt( f, "width" ), 3.5 )
 
+	def testOutputLPEs( self ) :
+
+		r = GafferScene.Private.IECoreScenePreview.Renderer.create(
+			"Arnold",
+			GafferScene.Private.IECoreScenePreview.Renderer.RenderType.SceneDescription,
+			self.temporaryDirectory() + "/test.ass"
+		)
+
+		r.output(
+			"test",
+			IECore.Display(
+				"beauty.exr",
+				"exr",
+				"lpe:C.*D.* RGB",
```

Comment:
Oh, right, we currently accept either Arnold naming convention ( "aov RGB" ) , or Gaffer naming convention ( "color aov" ).  Internally, I've ended up mostly just using Arnold naming convention to match the Arnold references, but I guess it makes sense that you would prefer we make things match the Gaffer naming convention instead.

The problem with either "color lpe:C.*" or "lpe C.*" is that neither offers a way to differentiate between the final Arnold data "myLpeName RGB" and "myLpeName RGBA".  I guess it would be somewhat reasonable to always use RGB for LPEs, but there could be legitimate cases where you want to use an LPE for the beauty, and then you would like to get an LPE aov with alpha.

Do we need to start by establishing a general Gaffer convention for aovs which include alpha ( even apart from LPEs )?  Data of "color4 aov"?  Add an Output parameter "includeAlpha" ?

If we had an Output parameter "includeAlpha" to control whether you actually want "lpeName RGB" or "lpeName RGBA", then putting "lpe C.*" in the data sounds fine to me.

Reply:
> Internally, I've ended up mostly just using Arnold naming convention to match the Arnold references, but I guess it makes sense that you would prefer we make things match the Gaffer naming convention instead.

That's definitely my preference, yes. In an ideal world, output definitions would be transferrable between renderers, especially since most of them are using OSL now. In practice I don't think we're there yet, because Arnold/PRMan support LPEs but 3Delight supports a separate `outputvariable()` closure, and Appleseed only has a few hardcoded AOVs. But I'd like to aim that way...

> The problem with either "color lpe:C." or "lpe C." is that neither offers a way to differentiate between the final Arnold data "myLpeName RGB" and "myLpeName RGBA". I guess it would be somewhat reasonable to always use RGB for LPEs, but there could be legitimate cases where you want to use an LPE for the beauty, and then you would like to get an LPE aov with alpha.

Ah yep, that makes sense. I do still like "lpe C." though, because it matches the existing "<type> <name>" syntax.

> Do we need to start by establishing a general Gaffer convention for aovs which include alpha ( even apart from LPEs )? Data of "color4 aov"? Add an Output parameter "includeAlpha" ?

3Delight (NSI) has gone with the "includeAlpha" approach, and it seems pretty reasonable to me. From a user perspective, I think a separate checkbox is preferable to mucking about with strings in the data field.

> If we had an Output parameter "includeAlpha" to control whether you actually want "lpeName RGB" or "lpeName RGBA", then putting "lpe C.*" in the data sounds fine to me.

I think we 're in agreement then. Although I suggest we just assume RGB for now, and then add the "includeAlpha" support later, so we can put more time/thought into making it work across all the renderer backends. There's already a tonne of stuff happening in this PR, and I don't think RGB-only LPEs is really a problem for a first release.